spec/sequencer_spec.rb

```diff
@@ -337,10 +331,13 @@
         described_class.new(sequence_table, opts)
       end
 
+      before do
+        expect(cache).to receive(:get).with(ident).and_yield
```

Comment:
This kind of confuses me but I think I'm probably just missing some RSpec knowledge. Is this an assertion that runs before the code under test is called?

Reply:
It's kind of a mock and an assertion at the same time. It mocks out the `#get` method on the `cache`object, but also states that the method **must** be called with the correct parameters or the test fails. The `and_yield` bit specifies it's return value, specifically it's `yield`s any block you pass the method

Reply:
https://relishapp.com/rspec/rspec-mocks/docs/basics/expecting-messages probably exlpains better than me